src/js/components/MultiSelect/MultiSelect.js

```diff
@@ -0,0 +1,680 @@
+import React, {
+  forwardRef,
+  isValidElement,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+
+import { controlBorderStyle, useKeyboard, useForwardedRef } from '../../utils';
+import { defaultProps } from '../../default-props';
+
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { CheckBox } from '../CheckBox';
+import { DropButton } from '../DropButton';
+import { Keyboard } from '../Keyboard';
+import { FormContext } from '../Form/FormContext';
+import { Text } from '../Text';
+
+import { MultiSelectContainer } from './MultiSelectContainer';
+import {
+  applyKey,
+  HiddenInput,
+  SelectTextInput,
+  StyledSelectDropButton,
+  calcValuedValue,
+  changeEvent,
+  getSelectIcon,
+  checkDisabled,
+  getIconColor,
+  getOptionValue,
+  getOptionLabel,
+  getDisplayLabelKey,
+  SelectOption,
+  DefaultSelectTextInput,
+} from '../Select/utils';
+import { MessageContext } from '../../contexts/MessageContext';
+import { MultiSelectPropTypes } from './propTypes';
+
+const StyledSelectBox = styled(Box)`
+  ${(props) => !props.callerPlain && controlBorderStyle};
+  ${(props) =>
+    props.theme.select &&
+    props.theme.select.control &&
+    props.theme.select.control.extend};
+  ${(props) => props.open && props.theme.select.control.open};
+`;
+
+StyledSelectDropButton.defaultProps = {};
+Object.setPrototypeOf(StyledSelectDropButton.defaultProps, defaultProps);
+
+const MultiSelect = forwardRef(
+  (
+    {
+      a11yTitle,
+      'aria-label': ariaLabel,
+      alignSelf,
+      children,
+      defaultValue,
+      disabled,
+      disabledKey,
+      dropAlign,
+      dropHeight,
+      dropProps,
+      dropTarget,
+      emptySearchMessage,
+      focusIndicator,
+      gridArea,
+      helpContent,
+      id,
+      icon,
+      labelKey,
+      limit,
+      margin,
+      messages,
+      name,
+      onBlur,
+      onChange,
+      onClick,
+      onClose,
+      onFocus,
+      onKeyDown,
+      onMore,
+      onOpen,
+      onSearch,
+      open: propOpen,
+      options: optionsProp,
+      placeholder,
+      plain,
+      replace,
+      searchPlaceholder,
+      size,
+      value: valueProp,
+      valueKey,
+      valueLabel,
+      visibleSelection = false,
```

Comment:
```suggestion
      showSelected = false,
```
```suggestion
      showSelection = false,
```

throwing out a couple other ideas for this prop name

Reply:
Good suggestions, I like `showSelectedInline` I think it provides the most clarity to what the prop is doing.